content/articles/encapsulation-in-c++/index.md

```diff
@@ -0,0 +1,132 @@
+#### Encapsulation in C++
+### Introduction to Encapsulation
+Encapsulation is the method of grouping data and features into a single part. This part is known as a **class**. It is done to protect straightforward access to the data. Encapsulation is one of the **Object-Oriented Programming (OOPs)** features.
+Construction of user-limited classes in **C++** supports encapsulation and data hiding. 
+### Table of Contents
+- [Introduction to Encapsulation](#introduction-to-encapsulation)
+- [Table of Contents](#table-of-contents)
+- [Encapsulation and Abstraction](#encapsulation-and-abstraction)
+- [Difference between Abstraction and Encapsulation](#difference-between-abstraction-and-encapsulation)
+- [Encapsulation vs Inheritance](#incapsulation-vs-inheritance)
+- [How Encapsulation is Implemented](#how-encapsulation-is-implemented)
+- [Why Encapsulation](#why-encapsulation)
+- [Types of Encapsulation](#types-of-encapsulation)
+- [The Role of Encapsulation](#the-role-of-encapsulation)
+- [Data Hiding in relation to Encapsulation](#data-hiding-in-relation-to-encapsulation)
+- [Roles of Access specifiers in Encapsulation](#roles-of-access-specifiers-in-encapsulation)
+- [Real Life Examples of Encapsulation](#real-Life-examples-of-encapsulation)
+- [Conclusion](#conlusion)
+- [Further Reading](#further-reading)
+### Encapsulation and Abstraction
+**Data Abstraction** is a process by which the programmer decides what data is to be shown in public; nevertheless, the practical execution of the Abstraction is nothing more than Encapsulation, which is accomplished through the use of access modifiers. 
+You may argue that Encapsulation is the Abstraction's implementation.
+### Difference between Abstraction and Encapsulation
+* Abstraction itself is a process that involves obtaining information, while Encapsulation, on the other side, involves methods of containing Information.
+* Abstraction is a technique for masking undesired data and revealing useful information. On the other hand, Encapsulation is the process of combining code and data into a unified unit to safeguard data from external view.
+* Encapsulation involves masking the internal details or mechanics of how an object does something. In contrast, abstraction allows you to concentrate on what the item does rather than how it does it.
+* Abstraction is achieved by using abstract classes and interfaces, while Encapsulation, is implemented by using access modifiers (Public, Protected, and Private).
+### Encapsulation vs. Inheritance 
+Encapsulation links a code to the data it distorts. Inheritance allows a class to inherit the attributes and functions of another class. Encapsulation is exemplified by an ATM that binds several denominations of banknotes and all the activities necessary to withdraw money. Inheritance may be seen in the classification of vehicles like cars, bikes, buses, trucks, etc.
```

Comment:
After this sentence you should add a space line, otherwise it will be like a big sentence. (do this with all sentences that need an extra space line.